### PR TITLE
Add basic batching to zed import

### PR DESCRIPTION
**Pull request by bryanhuhta**
This PR adds basic batching support for the `import` command, allowing `zed` to process large relationship files. Previously, `zed import` would tip over with ~9k relationship tuples (this is dependent on a few factors, like which DB driver is being used).

The general strategy presented here is to load all relationship tuples into memory (this behavior is unchanged), then in batches of 5000, serially import the tuples. This is by no means the most performant batching implementation, but this change will at least allow `zed` to import arbitrarily large files (provided the entire file can be initially stored in memory).

Related: https://github.com/authzed/zed/issues/94

### CODE REVIEW

File: cmd/zed/import.go [hunk 1]
```diff
@@ -140,13 +140,38 @@ func importRelationships(client *authzed.Client, relationships string) error {
...
+	totalBatches := int(float64(len(updates))/float64(batchSize)+0.5) + 1
+	batchCount := 1
+
+	for i := 0; i < len(updates); i += batchSize {
+		end := i + batchSize
+		if end > len(updates) {
+			end = len(updates)
+		}
+
+		request := &v1.WriteRelationshipsRequest{Updates: updates[i:end]}
```
Comment (ecordell) [inline cmd/zed/import.go:160]: You'd have a lot less to keep track of with something like:

```suggestion
        for i := 0; i < len(updates)/batchSize; i++ {
                request := &v1.WriteRelationshipsRequest{Updates: updates[i*batchSize : (i+1)*batchSize]}
```

File: cmd/zed/import.go [hunk 2]
```diff
@@ -140,13 +140,38 @@ func importRelationships(client *authzed.Client, relationships string) error {
 	if err := scanner.Err(); err != nil {
 		return err
 	}
+	log.Info().Int("count", len(relationshipUpdates)).Msg("parsed relationships")
 
-	request := &v1.WriteRelationshipsRequest{Updates: relationshipUpdates}
-	log.Trace().Interface("request", request).Msg("writing relationships")
-	log.Info().Int("count", len(relationshipUpdates)).Msg("importing relationships")
+	const importBatchSize = 5000
```
Comment (josephschorr) [inline cmd/zed/import.go:145]: Move const to outside of the function

File: cmd/zed/import.go [hunk 3]
```diff
@@ -140,13 +140,38 @@ func importRelationships(client *authzed.Client, relationships string) error {
...
+	return batchRequests(client, relationshipUpdates, importBatchSize)
+}
 
-	if _, err := client.WriteRelationships(context.Background(), request); err != nil {
-		return err
+func batchRequests(client *authzed.Client, updates []*v1.RelationshipUpdate, batchSize int) error {
+	// Number of batches required to import all relationships.
+	totalBatches := int(float64(len(updates))/float64(batchSize)+0.5) + 1
```
Comment (josephschorr) [inline cmd/zed/import.go:151]: Add a log for the number of requests to be made

File: cmd/zed/import.go [hunk 4]
```diff
@@ -140,13 +142,34 @@ func importRelationships(client *authzed.Client, relationships string) error {
...
-	request := &v1.WriteRelationshipsRequest{Updates: relationshipUpdates}
-	log.Trace().Interface("request", request).Msg("writing relationships")
-	log.Info().Int("count", len(relationshipUpdates)).Msg("importing relationships")
+	return batchRequests(client, relationshipUpdates, importBatchSize)
+}
 
-	if _, err := client.WriteRelationships(context.Background(), request); err != nil {
-		return err
+func batchRequests(client *authzed.Client, updates []*v1.RelationshipUpdate, batchSize int) error {
+	totalBatches := (len(updates) + batchSize - 1) / batchSize
```
Comment (bryanhuhta) [inline cmd/zed/import.go:151]: I changed the calculation for getting the total number of batches required. The last one was unintuitive.

File: cmd/zed/import.go [hunk 5]
```diff
@@ -140,13 +142,34 @@ func importRelationships(client *authzed.Client, relationships string) error {
...
+	log.Info().Int("batch_size", batchSize).Int("total_batches", totalBatches).Msg("batching relationship writes")
+
+	for i := 0; i < totalBatches; i++ {
```
Comment (bryanhuhta) [inline cmd/zed/import.go:154]: @ecordell Had a review (that's gone now? Maybe a ninja delete?) about the loop potentially being simpler. I re-tooled the loop to make it hopefully more straightforward.

File: cmd/zed/import.go [hunk 6]
```diff
@@ -140,13 +142,34 @@ func importRelationships(client *authzed.Client, relationships string) error {
...
 
-	request := &v1.WriteRelationshipsRequest{Updates: relationshipUpdates}
-	log.Trace().Interface("request", request).Msg("writing relationships")
-	log.Info().Int("count", len(relationshipUpdates)).Msg("importing relationships")
+	return batchRequests(client, relationshipUpdates, importBatchSize)
+}
 
-	if _, err := client.WriteRelationships(context.Background(), request); err != nil {
-		return err
+func batchRequests(client *authzed.Client, updates []*v1.RelationshipUpdate, batchSize int) error {
```
Comment (josephschorr) [inline cmd/zed/import.go:150]: Add a test for this?

**Comment by bryanhuhta** [comment]
closing in favor of https://github.com/authzed/zed/pull/120

**Pull request closed without merging**